validation-test/Reflection/reflect_nested_generic.swift

```diff
@@ -0,0 +1,117 @@
+
+// REQUIRES: reflection_test_support
+// REQUIRES: objc_interop
+// REQUIRES: executable_test
+// UNSUPPORTED: use_os_stdlib
+
+import SwiftReflectionTest
+
+enum Outer1 {
+  class C<T> {
+    enum Inner {
+      struct S<U> {
+	var u: U? = nil
+	var t: T? = nil
+      }
+    }
+  }
+}
+
+reflect(enum: Outer1.C<Void>.Inner.S<(Int,Int)>?.none)
+
+//CHECK: Reflecting an enum.
+//CHECK-NEXT: Instance pointer in child address space: 0x{{[0-9a-fA-F]+}}
+//CHECK-NEXT: Type reference:
+//CHECK-NEXT: (bound_generic_enum Swift.Optional
+//CHECK-NEXT:   (bound_generic_struct reflect_nested_generic.Outer1.C.Inner.S
+//CHECK-NEXT:     (tuple
+//CHECK-NEXT:       (struct Swift.Int)
+//CHECK-NEXT:       (struct Swift.Int))
+//CHECK-NEXT:     (bound_generic_class reflect_nested_generic.Outer1.C
+//CHECK-NEXT:       (tuple))))
+
+//CHECK: Type info:
+
+//X64-NEXT: (single_payload_enum size=19 alignment=8 stride=24 num_extra_inhabitants=0 bitwise_takable=1
+//X64-NEXT:   (case name=some index=0 offset=0
+//X64-NEXT:     (struct size=18 alignment=8 stride=24 num_extra_inhabitants=0 bitwise_takable=1
+//X64-NEXT:       (field name=u offset=0
+//X64-NEXT:         (single_payload_enum size=17 alignment=8 stride=24 num_extra_inhabitants=0 bitwise_takable=1
+//X64-NEXT:           (case name=some index=0 offset=0
+//X64-NEXT:             (tuple size=16 alignment=8 stride=16 num_extra_inhabitants=0 bitwise_takable=1
+//X64-NEXT:               (field offset=0
+//X64-NEXT:                 (struct size=8 alignment=8 stride=8 num_extra_inhabitants=0 bitwise_takable=1
+//X64-NEXT:                   (field name=_value offset=0
+//X64-NEXT:                     (builtin size=8 alignment=8 stride=8 num_extra_inhabitants=0 bitwise_takable=1))))
+//X64-NEXT:               (field offset=8
+//X64-NEXT:                 (struct size=8 alignment=8 stride=8 num_extra_inhabitants=0 bitwise_takable=1
+//X64-NEXT:                   (field name=_value offset=0
+//X64-NEXT:                     (builtin size=8 alignment=8 stride=8 num_extra_inhabitants=0 bitwise_takable=1))))))
+//X64-NEXT:           (case name=none index=1)))
+//X64-NEXT:       (field name=t offset=17
+//X64-NEXT:         (single_payload_enum size=1 alignment=1 stride=1 num_extra_inhabitants=0 bitwise_takable=1
+//X64-NEXT:           (case name=some index=0 offset=0
+//X64-NEXT:             (tuple size=0 alignment=1 stride=1 num_extra_inhabitants=0 bitwise_takable=1))
+//X64-NEXT:           (case name=none index=1)))))
+//X64-NEXT:   (case name=none index=1))
+
+//TODO:  X32
+
+//CHECK: Mangled name: $s22reflect_nested_generic6Outer1O1CC1SVy_yt_Si_SitGSg
+//CHECK-NEXT: Demangled name: Swift.Optional<reflect_nested_generic.Outer1.C<()>.S<(Swift.Int, Swift.Int)>>
+
+//CHECK: Enum value:
+//CHECK-NEXT: (enum_value name=none index=1)
+
+
+protocol P {}
+
+struct S1: P { var a: Int; var b: Int }
+struct S2: P { }
+
+struct Outer2 {
+  enum E<T: P> {
+    struct Inner {
+      enum F<U: P> {
```

Comment:
Changing this to a multi-payload Enum fails.  I think that's just because all the MPE fixes from #64852 haven't landed yet.